build-plugins/build-support/src/main/kotlin/aws/sdk/kotlin/gradle/plugins/artifactmetrics/AnalyzeMetricsTask.kt

```diff
@@ -0,0 +1,148 @@
+                    )
+
+                    val remoteMetrics = remoteMetricsFile.toMap()
+                    val remoteLines = remoteMetricsFile.readLines()
+                    val remoteHeaders = remoteLines[0].split(",").map { it.trim() }
+
+                    val localMetrics = metricsFile.get().asFile.toMap()
+                    val localLines = metricsFile.get().asFile.readLines()
+                    val localHeaders = localLines[0].split(",").map { it.trim() }
+
+                    val artifactNames = remoteHeaders.toSet() + localHeaders.toSet()
+                    val metrics = mutableMapOf<String, DataPoint>()
+
+                    var addedArtifact = false
+                    var removedArtifact = false
+                    var significantChange = false
+
+                    artifactNames.forEach { artifact ->
+                        val local = localMetrics[artifact] ?: 0
+                        val remote = remoteMetrics[artifact] ?: 0
+
+                        val diff = local - remote
+                        val percentage = buildString {
+                            if (remote == 0L || local == 0L) {
+                                append("NaN")
+
+                                if (remote == 0L) addedArtifact = true
+                                if (local == 0L) removedArtifact = true
+
+                                return@buildString
+                            }
+
+                            if (diff < 0) append("-")
+
+                            val percentageValue = (abs(diff).toDouble() / remote.toDouble() * 100).roundToInt()
+                            if (percentageValue > 5 && diff > 0) significantChange = true
+
+                            append(percentageValue)
+                            append("%")
+                        }
+
+                        metrics[artifact] =
+                            DataPoint(
+                                remote,
+                                local,
+                                diff,
+                                percentage,
+                            )
+                    }
+
+                    shouldFailWorkflow.get().asFile.writeText(significantChange.toString())
+
+                    val diffTable = buildString {
+                        appendLine("Affected Artifacts\n=")
+                        appendLine("| Artifact |Pull Request (bytes) | Latest Release (bytes) | Delta (bytes / percentage) |")
+                        appendLine("| -------- | ------------------- | ---------------------- | -------------------------- |")
+                        metrics.forEach { artifact ->
+                            if (artifact.value.diff != 0L) {
+                                appendLine("|${artifact.key}|${"%,d".format(artifact.value.localSize)}|${"%,d".format(artifact.value.latestReleaseSize)}|${"%,d".format(artifact.value.diff)} / ${artifact.value.percentage}|")
```

Comment:
**Style**: This line is particularly hard to read since it mixes string building, string interpolation, string formatting, and is longer than 120 chars. I suggest splitting this up into individual `append` calls for the sake of readability:

```kotlin
// cache formatters outside the loop
val sizeFmt = NumberFormat.getNumberInstance()
val pctFmt = NumberFormat.getPercentInstance()
…
metrics.forEach { artifact ->
    …
    append('|')
    append(artifact.key)
    append('|')
    append(sizeFmt.format(artifact.value.localSize))
    append('|')
    append(sizeFmt.format(artifact.value.latestReleaseSize))
    append('|')
    append(sizeFmt.format(artifact.value.diff))
    append('|')
    append(pctFmt.format(artifact.value.percentage))
    appendLine('|')
    …
}
```

Alternatively, you could use `listOf` and `joinToString` to do it slightly less verbosely:

```kotlin
metrics.forEach { artifact ->
    …
    val row = listOf(
        artifact.key,
        sizeFmt.format(artifact.value.localSize),
        sizeFmt.format(artifact.value.latestReleaseSize),
        sizeFmt.format(artifact.value.diff),
        pctFmt.format(artifact.value.percentage),
    ).joinToString(separator = "|", prefix = "|", postfix = "|")
    appendLine(row)
    …
}
```